### PR TITLE
feat: introduce vector type

### PR DESCRIPTION
**Pull request by zhongzc**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

Introduce vector type:

```SQL
> CREATE TABLE t (ts TIMESTAMP TIME INDEX, v VECTOR(3));

Affected Rows: 0

> INSERT INTO t VALUES
(1, "[1.0, 2.0, 3.0]"),
(2, "[4.0, 5.0, 6.0]"),
(3, "[7.0, 8.0, 9.0]");

Affected Rows: 3

> SELECT * FROM t;

+----------------------------+---------+
| ts                         | v       |
+----------------------------+---------+
| 1970-01-01 00:00:00.001000 | [1,2,3] |
| 1970-01-01 00:00:00.002000 | [4,5,6] |
| 1970-01-01 00:00:00.003000 | [7,8,9] |
+----------------------------+---------+
```

## Checklist

- [x] I have written the necessary rustdoc comments.
- [x] I have added the necessary unit tests and integration tests.
- [x] This PR requires documentation updates.

### CODE REVIEW

File: tests/cases/standalone/common/types/vector/vector.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+-- SQLNESS PROTOCOL MYSQL
+SELECT * FROM t2;
+
+-- SQLNESS PROTOCOL POSTGRES
+SELECT * FROM t2;
+
+DROP TABLE t;
+
+DROP TABLE t2;
```
Comment (waynexia) [inline tests/cases/standalone/common/types/vector/vector.sql:41]: nit
```suggestion
DROP TABLE t2;

```

File: tests/cases/standalone/common/types/vector/vector.sql
Comment (waynexia) [inline tests/cases/standalone/common/types/vector/vector.sql:1]: Can we add some computation in this test?

**Review by waynexia — APPROVED** [review]
I've skimmed through this, very concrete 💯

File: tests/cases/standalone/common/types/vector/vector.sql
Comment (zhongzc) [reply tests/cases/standalone/common/types/vector/vector.sql:1]: No computation yet, the next PR will do this

File: src/datatypes/src/schema/constraint.rs [hunk 2]
```diff
@@ -215,6 +215,17 @@ fn create_current_timestamp_vector(
     }
 }
 
+fn value_type_match(column_type: &ConcreteDataType, value_type: ConcreteDataType) -> bool {
+    match (column_type, value_type) {
+        (ct, vt) if ct.logical_type_id() == vt.logical_type_id() => true,
+        // Vector ans Json type is encoded as binary
```
Comment (fengjiachun) [inline src/datatypes/src/schema/constraint.rs:221]:
```suggestion
        // Vector and Json type is encoded as binary
```

File: Cargo.toml [hunk 3]
```diff
@@ -121,7 +121,8 @@ etcd-client = { version = "0.13" }
 fst = "0.4.7"
 futures = "0.3"
 futures-util = "0.3"
-greptime-proto = { git = "https://github.com/GreptimeTeam/greptime-proto.git", rev = "67bb1d52bc1241972c368657e658592b1be7ead3" }
+greptime-proto = { git = "https://github.com/zhongzc/greptime-proto.git", branch = "zhongzc/intro-vector-type" }
```
Comment (WenyXu) [inline Cargo.toml:124]: let's merge the proto first

**Review by WenyXu — APPROVED** [review]
Rest LGTM

File: src/datatypes/src/types/vector_type.rs [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+}
+
+/// Converts a vector type value to string
+/// for example: [1.0, 2.0, 3.0] -> "[1.0,2.0,3.0]"
+pub fn vector_type_value_to_string(val: &[u8], dim: u32) -> Result<String> {
+    if dim as usize * std::mem::size_of::<f32>() != val.len() {
+        return InvalidVectorSnafu {
+            msg: format!(
+                "Failed to convert Vector value to string: wrong byte size, expected {}, got {}",
+                dim as usize * std::mem::size_of::<f32>(),
```
Comment (fengjiachun) [inline src/datatypes/src/types/vector_type.rs:76]: Calculated twice (`dim as usize * std::mem::size_of::<f32>()`)

**Pull request merged**